interface/smart/register-app.php

```diff
@@ -0,0 +1,159 @@
+            function registerApp() {
+                let form = document.querySelector('form[name="app_form]');
+                let appRegister = {
+                    "application_type": "private"
+                    ,"redirect_uris": []
+                    ,"post_logout_redirect_uris": []
+                    ,"client_name": ""
+                    ,"token_endpoint_auth_method": "client_secret_post"
+                    ,"contacts": []
+                    ,"scope": "openid email phone address api:oemr api:fhir api:port api:pofh launch"
+                };
+                appRegister.client_name = document.querySelector('#appName').value;
+                let redirect_uri = document.querySelector("#redirectUri").value;
+                appRegister.redirect_uris.push(redirect_uri);
+                // not sure we need logout redirect right now
+                appRegister.post_logout_redirect_uris.push(document.querySelector("#logoutURI").value);
+                appRegister.contacts.push(document.querySelector("#contactEmail").value);
+
+                fetch(fhirRegistrationURL, {
+                    method: 'POST', // *GET, POST, PUT, DELETE, etc.
+                    headers: {
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify(appRegister) // body data type must match "Content-Type" header
+                }).then(result => {
+                    if (!result.ok) {
+                        return result.json().then(json => { throw json });
+                    }
+                    return result.json();
+                }).then(resultJSON => {
+                    console.log(resultJSON);
+                    document.querySelector(".apiResponse").classList.remove("hidden");
+                    document.querySelector(".errorResponse").classList.add("hidden");
+                    document.querySelector("#clientID").value = resultJSON.client_id;
+                })
+                .catch(error => {
+                    console.error(error);
+                    let msgText = error.message;
+                    if (!msgText) {
+                        msgText = JSON.stringify(error);
+                    }
+                    document.querySelector(".apiResponse").classList.add("hidden");
+                    document.querySelector(".errorResponse").classList.remove("hidden");
+                    document.querySelector("#errorResponseContainer").textContent = msgText;
+                });
+                return false;
+            }
+
+            window.addEventListener('load', function() {
+                document.querySelector('#submit').addEventListener('click', registerApp);
+            });
+        })(window, "<?php echo $fhirRegisterURL; ?>");
+    </script>
+</head>
+<body class="login">
+<form id="app_form" method="POST" autocomplete="off" action="../main/main_screen.php?auth=login&site=<?php echo attr($_SESSION['site_id']); ?>" target="_top">
```

Comment:
The action path above doesn't look right

Reply:
I can just remove the action path, its not even used as everything is handled via javascript.

Reply:
also i'd think we don't need $ignoreAuth for this form do we?

Reply:
I couldn't figure out why it was required, but I got a whitescreen for this page if $ignoreAuth wasn't here.  I assumed it had to do with something about it being public accessible like the login page is.  Maybe you or @bradymiller could chime in on why that was happening.

Reply:
Don't worry about it as i'll be in this form!